draft-ietf-httpbis-resumable-upload.md

```diff
@@ -203,11 +203,11 @@ The Upload Creation Procedure is intended for starting a new upload. A limited f
 
 This procedure is designed to be compatible with a regular upload. Therefore all methods are allowed with the exception of `GET`, `HEAD`, `DELETE`, and `OPTIONS`. All response status codes are allowed. The client is RECOMMENDED to use the `POST` method if not otherwise intended. The server MAY only support a limited number of methods.
 
-The request MUST include the `Upload-Token` header field ({{upload-token}}) which uniquely identifies an upload. The client MUST NOT reuse the token for a different upload. The request MUST NOT include the `Upload-Offset` header.
+The request MUST include the `Upload-Incomplete` header field ({{upload-incomplete}}). It MUST be set to true if the end of the request body is not the end of the upload. Otherwise, it MUST be set to false. This header field can be used for request identification by a server. The request MUST NOT include the `Upload-Offset` header.
```

Comment:
I think of `Upload-Incomplete` as an advanced feature for people who need to do chunked uploads. Using it to identify resumable upload feels a bit wrong. Also `Upload-Incomplete = false` is a double negative which can be confusing. If we go this route, I prefer flipping it to `Upload-Complete` instead. (Also don't forget to update the examples)

Reply:
You touch on two good points here. I abuse the Upload-Incomplete header for feature detection here. This may or may not be a smart choice. Maybe we use a more explicit header as mentioned in https://github.com/httpwg/http-extensions/issues/2295 and keep Upload-Incomplete optional.

Removing the double-negative also sounds good but maybe let's do this in a separate PR.